### PR TITLE
Rephrase the release highlight and capitalize the notes

### PR DESCRIPTION
**Pull request by zelinh**
Signed-off-by: Zelin Hao <zelinhao@amazon.com>

### Description
Some modifications on the 1.3.0 release notes:
- Rephrase the words in the release highlight section for ml-commons.
- Rename some of the titles to use formal name. (e.g. K-NN, SQL)
- Capitalize the first letter of each sentences for some parts.
 
### Issues Resolved
Continues on https://github.com/opensearch-project/opensearch-build/pull/1766
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-release-notes-1.3.0.md [hunk 1]
```diff
@@ -233,46 +233,46 @@ OpenSearch Dashboards [Release Notes](https://github.com/opensearch-project/Open
 * Changes the default policy of the visual editor to be an empty one ([#149](https://github.com/opensearch-project/index-management-dashboards-plugin/pull/149))
 
 
-### OpenSearch Knn
+### OpenSearch K-NN
```
Comment (gaiksaya) [inline release-notes/opensearch-release-notes-1.3.0.md:236]: It's small k. k-NN :)
Ref: https://github.com/opensearch-project/k-NN

Comment (zelinh) [reply release-notes/opensearch-release-notes-1.3.0.md:236]: All of our plugin repos are using lower case for the first letter. (e.g. alerting, security) But here we capitalize the first letters of them. (e.g. OpenSearch Alerting, OpenSearch Security). I believe we should follow the same convention for `k-NN` in this case. 
Or in another case, I know k-NN is a specific statistics term. We should use its scientific term then. WDYT?

Comment (zelinh) [reply release-notes/opensearch-release-notes-1.3.0.md:236]: I just saw from previous notes. We used `k-NN` all the time. I will changed the name to be the same with them. :D

Comment (gaiksaya) [reply release-notes/opensearch-release-notes-1.3.0.md:236]: I would suggest to stay with `k-NN` following https://github.com/opensearch-project/opensearch-build/blob/main/release-notes/opensearch-release-notes-1.2.0.md#opensearch-k-nn

Comment (zelinh) [reply release-notes/opensearch-release-notes-1.3.0.md:236]: Changed. Thanks!

File: release-notes/opensearch-release-notes-1.3.0.md [hunk 2]
```diff
@@ -3,7 +3,7 @@
 ## Release Highlights
 
 ### OpenSearch Ml Commons
-* ml-commons now officially becomes part of the OpenSearch 1.3.0 release.
+* The new ML Commons plugin empowers users to train and apply machine learning models as a part of the OpenSearch 1.3.0 release.
```
Comment (gaiksaya) [inline release-notes/opensearch-release-notes-1.3.0.md:6]: @ylwu-amzn @Naarcha-AWS @stockholmux Please review this release highlight for ml-commons.
Thanks!

**Review by gaiksaya — APPROVED** [review]

File: release-notes/opensearch-release-notes-1.3.0.md (hunk 2 above)
Comment (ylwu-amzn) [reply release-notes/opensearch-release-notes-1.3.0.md:6]: LGTM. Thanks @gaiksaya

**Review by peterzhuamazon — APPROVED** [review]

**Pull request merged**